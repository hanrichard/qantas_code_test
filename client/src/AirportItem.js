import React from 'react';
import { Link } from 'react-router-dom';

const AirportItem = props => {
    return (
        <div>
            <Link to={`/airport/${props.airport.airportCode}`}>
                <span>{props.airport.airportName}</span>
                <span>{props.airport.country.countryName}</span>
            </Link>
        </div>
    );
};

export default AirportItem;
