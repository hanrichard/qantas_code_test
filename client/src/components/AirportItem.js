import React from 'react';
import { Link } from 'react-router-dom';
import Card from '@material-ui/core/Card';
import styled from 'styled-components';
import componentStyle from './AirportItem.style';
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';

const Wrapper = styled.div`
    ${componentStyle}
`;
const AirportItem = props => {
    return (
        <Wrapper>
            <Card className="AirportList__card">
                <Link to={`/airport/${props.airport.airportCode}`} className="AirportList__link">
                    <span>
                        <span className="AirportList__link-airportName">{props.airport.airportName}</span>-
                        <span className="AirportList__link-countryName">{props.airport.country.countryName}</span>
                    </span>
                    <ArrowForwardIosIcon />
                </Link>
            </Card>
        </Wrapper>
    );
};

export default AirportItem;
