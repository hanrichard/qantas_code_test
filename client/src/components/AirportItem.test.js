import React from 'react';
import { shallow, configure } from 'enzyme';
import AirportItem from './AirportItem';
import Adapter from 'enzyme-adapter-react-16';
import Card from '@material-ui/core/Card';
import Renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

function setup() {
    const props = {
        airport: {
            airportName: 'syd',
            country: {
                countryName: 'aus',
            },
            airportCode: 'aaa',
        },
    };

    const shallowWrapper = shallow(<AirportItem {...props} />);

    return {
        props,
        shallowWrapper,
    };
}

describe('Airport item', () => {
    const { shallowWrapper } = setup();
    it('should match snapshot', () => {
        const airportItem = Renderer.create(`<BrowserRouter>shallowWrapper</BrowserRouter>`).toJSON();
        expect(airportItem).toMatchSnapshot();
    });
});

describe('Airport item component', () => {
    it('should render self and its subcomponents', () => {
        const { shallowWrapper } = setup();

        expect(shallowWrapper.find(Card)).toHaveLength(1);
        expect(shallowWrapper.text().includes('syd')).toBe(true);
        expect(shallowWrapper.text().includes('aus')).toBe(true);
        expect(shallowWrapper.find('.AirportList__card')).toHaveLength(1);
        expect(shallowWrapper.find('.AirportList__link')).toHaveLength(1);
        expect(shallowWrapper.find('.AirportList__link-airportName')).toHaveLength(1);
        expect(shallowWrapper.find('.AirportList__link-countryName')).toHaveLength(1);
    });
});
