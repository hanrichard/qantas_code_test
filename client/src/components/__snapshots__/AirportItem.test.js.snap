// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airport item should match snapshot 1`] = `"<BrowserRouter>shallowWrapper</BrowserRouter>"`;
